370-auth-ui/oidc.md

```diff
@@ -0,0 +1,341 @@
+
+#### ui_locales
+
+No difference from the spec.
+
+#### id_token_hint
+
+No difference from the spec, for `prompt=none` case.
+
+#### login_hint
+
+It is not supported.
+
+#### acr_values
+
+It is not supported.
+
+#### code_challenge_method
+
+Only `S256` is supported. `plain` is not supported. The reason is the existing implementation of SSO is using `S256` already. Since it defaults to plain, it is required and the value must be `S256`.
+
+#### code_challenge
+
+No difference from the spec.
+
+#### Example of valid Authentication Request
+
+`response_type=code&scope=openid&client_id=CLIENT_ID&redirect_uri=REDIRECT_URI&code_challenge_method=S256&code_challenge=CODE_CHALLENGE&ui_locales=en`
+
+### Token Request
+
+The following sessions list out the request parameters and their difference from the spec.
+
+#### grant_type
+
+Either `authorization_code` or `refresh_token`
+
+#### code
+
+No difference from the spec.
+
+#### redirect_uri
+
+No difference from the spec.
+
+#### client_id
+
+It is always required. The value must be a valid Skygear API Key.
+
+#### code_verifier
+
+No difference from the spec.
+
+#### Example of valid Token Request
+
+`grant_type=authorization_code&code=AUTHORIZATION_CODE&redirect_uri=REDIRECT_URI&client_id=API_KEY&code_verifier=CODE_VERIFIER`
+
+### Token Response
+
+#### id_token
+
+No difference from the spec.
+
+#### token_type
+
+It is always the value `bearer`.
+
+#### access_token
+
+No difference from the spec.
+
+#### expires_in
+
+No difference from the spec.
+
+#### refresh_token
+
+No difference from the spec. Absent in [SPA](#case-2-single-page-web-app-spa) flow.
+
+#### scope
+
+It is always absent.
+
+#### Example of Token Response
+
+```json
+{
+  "token_type": "bearer",
+  "access_token": "access.token",
+  "refresh_token": "refresh.token"
+}
+```
+
+## Configurations
+
+### Sample configuration
+
+```yaml
+app_config:
+  clients:
+  - client_id: XXX
+    client_secret: XXX
+    session_lifetime: 86400
+    access_token_lifetime: 1800
+    session_idle_timeout_enabled: true
+    session_idle_timeout: 300
+    client_name: Mobile App
+    redirect_uris:
+    - "https://example.com"
+    logo_uri: "https://example.com/logo.png"
+    grant_types:
+    - "authorization_code"
+    - "refresh_token"
+    response_types:
+    - "code"
+```
+
+### Parameters
+
+#### OIDC client Metadata
+
+Some parameters are defined in OIDC, see [ClientMetadata](https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata).
+
+They are
+
+- `client_name`
+- `redirect_uris`
+- `login_uri`
+- `grant_types`
+- `response_types`
+
+#### Skygear-specific
+
+- `client_id`: OIDC client id.
+- `client_secret`: OIDC client secret, used in RP integration.
+- `access_token_lifetime`: OIDC access token lifetime in seconds, default to 1800.
+- `session_lifetime`: OIDC refresh token lifetime in seconds, default to max(access_token_lifetime, 86400). Must greater than or equal to `access_token_lifetime`.
```

Comment:
I think we don't need configuration of session here? Since there is no concept of session in OIDC, we should instead have configuration of refresh token here?

Reply:
Yes, you are right. I changed the name back to `refresh_token_lifetime`, that would be clearer.